conductr_cli/bndl_main.py

```diff
@@ -0,0 +1,597 @@
+    digest.update(refs_data)
+
+    refs_digest = digest.hexdigest()
+
+    return {
+        'config': oci_config_data,
+        'config_digest': oci_config_digest,
+        'manifest': oci_manifest_data,
+        'manifest_digest': oci_manifest_digest,
+        'refs': refs_data,
+        'refs_digest': refs_digest
+    }
+
+
+def oci_image_unpack(destination, data, is_dir):
+    temp_dir = tempfile.mkdtemp()
+
+    try:
+        if is_dir:
+            shutil.copytree(data, os.path.join(temp_dir, 'image'))
+        else:
+            data.extractall(temp_dir)
+
+        for base, dirs, files in os.walk(temp_dir):
+            if 'oci-layout' in files or 'refs' in dirs:
+                os.renames(base, destination)
+                break
+    finally:
+        if os.path.isdir(temp_dir):
+            shutil.rmtree(temp_dir)
+
+
+def file_write_bytes(path, bs):
+    with open(path, 'wb') as file:
+        file.write(bs)
+
+
+def docker_parse_image_name(name):
+    image_tag = name.split(':')
+
+    if len(image_tag) != 2:
+        raise ValueError('Invalid tag format')
+
+    return os.path.basename(image_tag[0]), image_tag[1]
+
+
+def docker_image_name_matches(wanted_name, wanted_tag, image_name):
+    return \
+        (wanted_tag is None or image_name.endswith(':{}'.format(wanted_tag))) and \
+        (wanted_name is None or os.path.basename(image_name).startswith('{}:'.format(wanted_name)))
+
+
+def docker_unpack(args, destination, data, is_dir):
+    temp_dir = tempfile.mkdtemp()
+
+    try:
+        os.makedirs(os.path.join(destination, 'blobs/sha256'))
+        os.makedirs(os.path.join(destination, 'refs'))
+
+        file_write_bytes(os.path.join(destination, 'oci-layout'), '{"imageLayoutVersion": "1.0.0"}'.encode('UTF-8'))
+
+        layers_to_digests = {}
+        digests = {}
+        sizes = {}
+
+        def handle_entry(name, fileobj, isdir, isfile):
+            parent_dir = os.path.dirname(name)
+            immediate_parent_dir = os.path.basename(parent_dir)
+
+            if parent_dir != '':
+                os.makedirs(os.path.join(temp_dir, 'layers', parent_dir), exist_ok=True)
+
+            file_name = os.path.basename(name)
+
+            if isdir:
+                os.makedirs(os.path.join(temp_dir, 'layers', file_name))
+            elif isfile and parent_dir == '':
+                with open(os.path.join(temp_dir, file_name), 'wb') as dest:
+                    shutil.copyfileobj(fileobj, dest)
+            elif isfile and file_name == 'layer.tar':
+                dest_path = os.path.join(temp_dir, 'layers', parent_dir, 'data')
+
+                with open(dest_path, 'wb') as dest_file:
+                    dest_file_digest = DigestReaderWriter(dest_file)
+
+                    # TODO investigate if below is still the case
+                    # bundles are packaged into zips, so we don't want to compress, but OCI tooling
+                    # as of 2017-03-14 is broken for plain tar files; they must be gzip
```

Comment:
Yeah, this will need to be investigated some time in several months when OCI tooling has matured, specifically `umoci` and `oci-create-runtime-bundle`. With that caveat, what's the best action to take? I see one of A) leave as-is, B) leave as-is, but also open an issue, and C) remove comment, open an issue.